### PR TITLE
Add 30s cap time for tag tests

### PR DESCRIPTION
**Pull request by stisac**
*Issue #, if available:* https://sim.amazon.com/issues/AML-109456

*Description of changes:* Limit the max runtime for tag tests to 30s

*Testing done:*

tests/integ/sagemaker/lineage/test_action.py .......                                                                                                                                                                                [ 23%]
tests/integ/sagemaker/lineage/test_artifact.py .........                                                                                                                                                                            [ 53%]
tests/integ/sagemaker/lineage/test_association.py ....                                                                                                                                                                              [ 66%]
tests/integ/sagemaker/lineage/test_context.py .......                                                                                                                                                                               [ 90%]
tests/integ/sagemaker/lineage/test_dataset_artifact.py .                                                                                                                                                                            [ 93%]
tests/integ/sagemaker/lineage/test_endpoint_context.py .                                                                                                                                                                            [ 96%]
tests/integ/sagemaker/lineage/test_model_artifact.py .                                                                                                                                                                              [100%]

### CODE REVIEW

**Review by danabens — DISMISSED** [review]

File: setup.py [hunk 1]
```diff
@@ -67,6 +67,7 @@ def read_version():
         "pytest-cov",
         "pytest-rerunfailures",
         "pytest-xdist",
+        "pytest-timeout",
```
Comment (metrizable) [inline setup.py:70]: nit: alphabetize?

```suggestion
        "pytest-timeout",
        "pytest-xdist",
```

File: tests/integ/sagemaker/lineage/test_action.py [hunk 2]
```diff
@@ -90,12 +94,14 @@ def test_tag(action_obj, sagemaker_session):
         )["Tags"]
         if actual_tags:
             break
+        time.sleep(1)
```
Comment (metrizable) [inline tests/integ/sagemaker/lineage/test_action.py:97]: probably can increase sleep? it's not going to save much time to check every 1s and by increasing to, say, 5s, you reduce call volumes (and potential throttling) by a factor 5.

Comment (stisac) [reply tests/integ/sagemaker/lineage/test_action.py:97]: Good point, will update. Thank you

File: setup.py (hunk 1 above)
Comment (stisac) [reply setup.py:70]: Yep, will alphabetize. Thank you

**Review by metrizable — APPROVED** [review]
thanks for the contribution!

**Review by danabens — APPROVED** [review]

**Pull request merged**